restapi/src/main/java/io/stargate/web/resources/ResourceUtils.java

```diff
@@ -0,0 +1,21 @@
+package io.stargate.web.resources;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ResourceUtils {
+
+  public static Map<String, Object> readJson(String payload, ObjectMapper mapper) {
+    Map<String, Object> requestBody;
+    try {
+      requestBody = mapper.readValue(payload, new TypeReference<HashMap<String, Object>>() {});
```

Comment:
While construction of `TypeReference` is cheap, resolving that to `JavaType` in Jackson has some overhead. Nothing drastic, but if we have an `ObjectMapper`, we could streamline things by pre-constructing type as well as creating `ObjectReader` (which pre-fetches deserializer to use):

```
ObjectReader MAP_READER = mapper.readerFor(mapper.constructType(new TypeReference<...>() { }));

....

Map<String, Object> requestBody = MAP_READER.readValue(payload);
```

I doubt this has measurable effect in grand scheme of things but is also not much more complicated way.

Reply:
Well that's cool. I'll make the change